### DISCUSSION TITLE
Questions regarding recursivity

### DISCUSSION THREAD

**Discussion by macheitor** [dirkpetersen/froster#33] [General]
- Recursivity in archiving
  - How deep the archiving tree usually is?
  - When storing a folder & subfolders there could be thousands of files, but how many subfolders would be the average?
  - I'm thinking to "get rid" of recursive attribute in froster-archives.json. The reason is that is complicates all the logic on other commands. ```froster archive --recursive /folder/to/archive```will continue working as always but it will make an entry to froster-archives.json for every archived folder. Does that make sense?

- Shall we allow user to mount recursively?
- Is it useful for a user to mount in a mountpoint different than the actual folder?

**Comment by macheitor** [comment]
When you mount, you always "do it recursively".
- When restoring, should we allow users to restore recursively a subdir of a recursive archived parent folder?

@dirkpetersen

**Comment by dirkpetersen** [comment]
> * Recursivity in archiving
>   
>   * How deep the archiving tree usually is?
>   * When storing a folder & subfolders there could be thousands of files, but how many subfolders would be the average?

The folder tree i am working with is about 7-8 folders deep and has a perhaps about 100 million folders

>   * I'm thinking to "get rid" of recursive attribute in froster-archives.json. The reason is that is complicates all the logic on other commands. `froster archive --recursive /folder/to/archive`will continue working as always but it will make an entry to froster-archives.json for every archived folder. Does that make sense?

Yes, that would actually be ideal any make restores easier. But I am concerned about scalability and also readability of the froster-archives.json which could have tens of thousands of folders quickly . A corner case that is not well tested is currently 

. someone archives /folder1/folder2/folder3 recursively and later someone else archives /folder1/folder2 non-recursively. The restores for these 2 cases are not well tested 

> * Shall we allow user to mount recursively?

Yes

> * Is it useful for a user to mount in a mountpoint different than the actual folder?

Rarely, but in some cases people might have restored once then make a change and might want to restore again without affecting the changed data, rare chase

**Comment by dirkpetersen** [comment]
> * When restoring, should we allow users to restore recursively a subdir of a recursive archived parent folder?

Yes, that is required as some people could restore huge tree but then want to get only a subset back. It is currently implemented but perhaps not working as expected ?
